src/game.rs

```diff
@@ -966,6 +966,117 @@ impl PostFlopGame {
+    /// board-pairing turns) which we cannot do while building an action tree.
+    pub fn remove_lines(&mut self, lines: &Vec<Vec<Action>>) -> Result<(), String> {
+        if self.state <= State::Uninitialized {
+            return Err("Game is not successfully initialized".to_string());
+        } else if self.state == State::MemoryAllocated {
+            return Err("Game has already been allocated".to_string());
+        }
+
+        for line in lines.iter() {
+            let mut root = self.root();
+            let info = self.remove_line_recursive(&mut root, line)?;
+            self.misc_memory_usage -= info.memory_usage_nodes;
+            self.num_storage_actions -= info.num_storage_actions;
+            self.num_storage_chances -= info.num_storage_chances;
+        }
+        Ok(())
+    }
+
+    /// Recursively remove a line from a PostFlopGame tree
+    fn remove_line_recursive(
+        &self,
+        node: &mut PostFlopNode,
+        line: &[Action],
+    ) -> Result<BuildTreeInfo, String> {
+        if line.is_empty() {
+            return Err("Empty line".to_string());
+        }
+
+        if node.is_terminal() {
+            return Err("Unexpected terminal node".to_string());
+        }
+
+        let action = line[0];
+        let search_result = node.actions.binary_search(&action);
+        if search_result.is_err() {
+            return Err(format!("Action does not exist: {action:?}"));
+        }
+
+        let index = search_result.unwrap();
+        if line.len() > 1 {
+            let result = self.remove_line_recursive(&mut node.children[index].lock(), &line[1..]);
+            if line.len() == 2 {}
+            return result;
+        }
+
+        // Remove action/children at index. To do this we must
+        // 1. compute the storage space required by the tree rooted at action index
+        // 2. remove children and actions
+        // 3. re-define num_elements and num_elements_aux after we remove children and actions
+
+        // STEP 1
+        let mut info = BuildTreeInfo {
+            memory_usage_nodes: mem::size_of::<PostFlopNode>() as u64,
+            num_storage_actions: 0,
+            num_storage_chances: 0,
+        };
+
+        let node_to_remove = &*node.children.get(index).unwrap().lock();
+        self.calculate_removed_line_info_recursive(node_to_remove, &mut info);
+
+        // STEP 2
+        node.actions.remove(index);
+        node.children.remove(index);
+
+        node.actions.shrink_to_fit();
+        node.children.shrink_to_fit();
+
+        // STEP 3
+        if node.is_chance() {
+            node.num_elements = match node.cfvalue_storage(PLAYER_OOP as usize) {
+                CfValueStorage::None => 0,
+                CfValueStorage::Sum => self.num_private_hands(PLAYER_OOP as usize),
+                CfValueStorage::All => {
+                    node.num_actions() * self.num_private_hands(PLAYER_OOP as usize)
+                }
+            };
```

Comment:
Removing a chance action is somewhat misleading, so I do not want to support it. I prefer returning an error rather than processing it.

Reply:
Okay, I added an error check above. I agree that it shouldn't be supported